api/src/opentrons/protocol_runner/run_orchestrator.py

```
@@ -334,3 +348,27 @@ def get_robot_type(self) -> RobotType:
     def get_deck_type(self) -> DeckType:
         """Get engine deck type."""
         return self._protocol_engine.state_view.config.deck_type
+
+    async def command_generator(self) -> AsyncGenerator[str, None]:
+        """Yield next command to execute."""
+        while True:
+            try:
+                # TODO(tz, 6-26-2024): avoid using private accessor in a follow up pr.
+                command_id = await self._protocol_engine._state_store.wait_for(
+                    condition=self._protocol_engine.state_view.commands.get_next_to_execute
+                )
+                # Assert for type hinting. This is valid because the wait_for() above
+                # only returns when the value is truthy.
+                assert command_id is not None
+                yield command_id
+            except RunStoppedError:
+                # There are no more commands that we should execute, either because the run has
+                # completed on its own, or because a client requested it to stop.
+                break
+
+    @staticmethod
+    def _map_parse_mode_to_python_parse_mode(parse_mode: ParseMode) -> PythonParseMode:
+        if parse_mode == ParseMode.NORMAL:
```

I can see the future and in the future we have added a new `ParseMode` and we are extremely confused about why it's not getting passed in to the protocol orchestrator. Please make this raise an error if it gets an unexpected `ParseMode`